src/vcpkg/commands.add.cpp

```diff
@@ -41,8 +46,69 @@ namespace vcpkg::Commands
             {
                 Checks::exit_with_message(VCPKG_LINE_INFO, "add port requires an active manifest file.\n");
             }
+            const std::vector<ParsedQualifiedSpecifier> specs = Util::fmap(args.command_arguments, [&](auto&& arg) {
+                ExpectedS<ParsedQualifiedSpecifier> value = parse_qualified_specifier(std::string(arg));
+                if (value)
+                {
+                    if (!value.value_or_exit(VCPKG_LINE_INFO).triplet)
+                    {
+                        return value.value_or_exit(VCPKG_LINE_INFO);
+                    }
+                    print2(Color::error,
+                           "Error: Triplet expressions are not allowed here. Use the --triplet parameter when running "
+                           "`vcpkg install`.\n");
+                }
+                else
+                {
+                    print2(Color::error, value.error());
+                }
+                Checks::exit_fail(VCPKG_LINE_INFO);
+            });
+            const auto& manifest_path = paths.get_manifest_path().value_or_exit(VCPKG_LINE_INFO);
+            auto maybe_manifest_scf = SourceControlFile::parse_manifest_object(manifest_path, *manifest);
+            if (!maybe_manifest_scf)
+            {
+                print_error_message(maybe_manifest_scf.error());
+                print2("See ", docs::manifests_url, " for more information.\n");
```

Comment:
Same here with localization;

```suggestion
                msg::println(AddSeeManifestsUrl, msg::url = docs::manifests_url);
```

and then add `DECLARE_MSG_ARG(url)` to messages.h with all the other message arg declarations.